src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -90,8 +90,10 @@ private static boolean isValidPath(Path filePath) {
      * Saves all data to this storage file.
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
+     * @throws FileNotFoundException if file is not found during runtime
+     * 
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, FileNotFoundException {
```

Comment:
Your current implementation does not check if the file exists to throw an exception. You must implement that part. 
The best place to implement it is in the save() method. 
The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.